### ISSUE TITLE
Override Finalizer

### ISSUE THREAD

**Issue by thansen24** [nats-io/nats.net.v1#357]
If I run the following sample code:

```
private static void Main(string[] args)
{
    const string _natsUrl = "127.0.0.1:4221";

    var connectionFactory = new NATS.Client.ConnectionFactory();

    var connection = connectionFactory.CreateConnection(_natsUrl);

    connection.Publish("PubSubTesting", ASCIIEncoding.UTF8.GetBytes("Testing!!"));
}
```

My console app will hang and not exit.  If I change my code to dispose the connection object after Publish it will work.  I assume something needs to override the finalizer and take some action to clean things up more gracefully in this case?

**Comment by watfordgnf** [comment]
This was fixed in #354 and should be in the next release.

**Comment by watfordgnf** [comment]
Duplicate of #353
